tracing-flame/src/lib.rs

```diff
@@ -0,0 +1,396 @@
+        let path = path.as_ref();
+        let file = File::create(path)
+            .map_err(|source| Kind::CreateFile {
+                path: path.into(),
+                source,
+            })
+            .map_err(Error)?;
+        let writer = BufWriter::new(file);
+        let layer = Self::new(writer);
+        let guard = layer.flush_on_drop();
+        Ok((layer, guard))
+    }
+}
+
+impl<S, W> Layer<S> for FlameLayer<S, W>
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+    W: Write + 'static,
+{
+    fn new_span(&self, _: &Attributes<'_>, id: &Id, ctx: Context<'_, S>) {
+        let samples = self.time_since_last_event();
+
+        let first = ctx.span(id).expect("expected: span id exists in registry");
+        let parents = first.from_root();
+
+        let mut stack = String::new();
+
+        stack.push_str("tracing");
+
+        for parent in parents {
+            stack.push_str("; ");
+            write(&mut stack, parent).expect("expected: write to String never fails");
+        }
+
+        write!(&mut stack, " {}", samples.as_nanos())
+            .expect("expected: write to String never fails");
+
+        let _ = writeln!(*self.out.lock().unwrap(), "{}", stack);
+    }
+
+    // #[allow(clippy::needless_return)]
+    fn on_close(&self, id: Id, ctx: Context<'_, S>) {
+        let panicking = std::thread::panicking();
+        macro_rules! expect {
+            ($e:expr, $msg:literal) => {
+                if panicking {
+                    return;
+                } else {
+                    $e.expect($msg)
+                }
+            };
+            ($e:expr) => {
+                if panicking {
+                    return;
+                } else {
+                    $e.unwrap()
+                }
+            };
+        }
+
+        let samples = self.time_since_last_event();
+        let first = expect!(ctx.span(&id), "expected: span id exists in registry");
+        let parents = first.from_root();
+
+        let mut stack = String::new();
+        stack.push_str("tracing; ");
+
+        for parent in parents {
+            expect!(
+                write(&mut stack, parent),
+                "expected: write to String never fails"
+            );
+            stack.push_str("; ");
+        }
+
+        expect!(
+            write(&mut stack, first),
+            "expected: write to String never fails"
+        );
+        expect!(
+            write!(&mut stack, " {}", samples.as_nanos()),
+            "expected: write to String never fails"
+        );
+
+        let _ = writeln!(*expect!(self.out.lock()), "{}", stack);
+    }
+}
+
+impl<S, W> FlameLayer<S, W>
+where
+    S: Subscriber + for<'span> LookupSpan<'span>,
+    W: Write + 'static,
+{
+    fn time_since_last_event(&self) -> Duration {
+        let now = Instant::now();
+
+        let prev = LAST_EVENT.with(|e| {
+            let prev = e.get();
+            e.set(now);
+            prev
+        });
+
+        now - prev
```

Comment:
Hmm, we're now [panicking here on CI](https://github.com/tokio-rs/tracing/pull/631/checks?check_run_id=572548666#step:4:626). 

It looks like bcea4a694daffeed5c69b08de77eacc4da8bcebc removed the START timestamp initialization when constructing a `FlameLayer`: https://github.com/tokio-rs/tracing/pull/631/commits/bcea4a694daffeed5c69b08de77eacc4da8bcebc#diff-349fc70c78b8d22b76bf83034759ccaaL211
that means that `START` is only initialized when a thread accesses `LAST_EVENT` for the first time, since its initializer references `START`: 
https://github.com/tokio-rs/tracing/blob/bcea4a694daffeed5c69b08de77eacc4da8bcebc/tracing-flame/src/lib.rs#L146-L148

I think we need to put some code that references `START` back in `FlameLayer::new`